app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt

```diff
@@ -158,25 +160,30 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
 
     private fun getChangeListenerForDeviceName(): Preference.OnPreferenceChangeListener {
         return Preference.OnPreferenceChangeListener { _, newValue ->
-            // Optimistically set the device name to what user requested.
-            val deviceNameKey = context!!.getPreferenceKey(R.string.pref_key_sync_device_name)
-            val preferenceDeviceName = findPreference<Preference>(deviceNameKey)
-            preferenceDeviceName?.summary = newValue as String
-
-            // This may fail, and we'll have a disparity in the UI until `updateDeviceName` is called.
-            lifecycleScope.launch(IO) {
-                try {
-                    accountManager.authenticatedAccount()?.let {
-                        it.deviceConstellation().setDeviceNameAsync(newValue)
+            // The network request requires a nonempty string, so don't persist any changes if the user inputs one.
+            if (newValue.toString().trim() == "") {
+                Toast.makeText(context, "Device name cannot be empty.", Toast.LENGTH_LONG).show()
```

Comment:
This string should be placed in our [strings.xml](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/res/values/strings.xml) file so it can be localized 😄 

Then we can reference it here with `context.getString(R.string.ID_OF_STRING)`

Reply:
Toasts aren't recommended for much more than debugging because they stop appearing if the user disables notifications for our app. Also, if you create Toasts in a loop they can continue for hours outside of the app, as happened to me with LockWise.

Reply:
@colintheshots what would you recommend? Would a Snackbar be good?

Reply:
Yes, a Snackbar will appear in app whether or not the user has notifications enabled. It's a good option for these quick, context-sensitive notifications. Toasts appear inside or outside the app and as such, require notifications access. Toasts can also appear on the wrong screen due if the user browses away.